modules/data-streams/src/main/java/org/elasticsearch/datastreams/action/TransportUpdateDataStreamMappingsAction.java

```diff
@@ -143,8 +144,9 @@ private void updateSingleDataStream(
             );
             return;
         }
+        ProjectId projectId = projectResolver.getProjectId();
```

Comment:
This is a pretty small nit, but we usually prefer to resolve the project ID only once, to avoid any confusion whatsoever (and to save a thread context lookup, although that's pretty minor).

Reply:
I'm only getting the project ID once, right? Where is the other time?

Reply:
Ah, ok. I've changed that now. I never thought of projectId as something I need to be careful about accessing. It's possible there are other places where I'm accessing it frequently.